chapter09/red/exercise2.pl

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+my ( $pattern_repo ) = @ARGV;
+if ( !defined $pattern_repo || !-r $pattern_repo ) {
+  print "Please supply a file containing line delimited regular expressions.\n";
+  exit 0;
+}
+
+open my $repo, '<', $pattern_repo or die "Failed to read pattern repository: $!";
+
+my @patterns = eval { map { chomp; qr/$_/ } <$repo> };
+
+die "Invalid regular expression: $@" if $@;
```

Comment:
Yup. This wasn't really specified in the question, but I decided an invalid pattern repository would be some kind of stop-the-world error the user would need to fix.